### ISSUE TITLE
Auto dump log on errors

### ISSUE THREAD

**Issue by AbdealiJK** [coala-analyzer/coala#1862]
This is something I think would be very useful.

The feature request is that if coala fails, we auto dump a `.coala-log` or `coala-log.txt` file in the current directory where it's being run from. Why ?

There have been many instances where I find that coala failed due to an unusual error. Specifically when I'm running coala and using the editor/auto apply patch in various places. I get the "Sorry, something went wrong, this is a bug" error but I am unable to reproduce it later :/

It's frustrating, and we should have a "dump all error messages if it bugged out" because the issues aren't always reproducible.

Other notes: ros, pip, and npm have this.

**Comment by sils1297** [comment]
Neat, sounds interesting and not particularly hard to do either, we have the exception catching in one routine somewhere where this could be done, right?

**Comment by AbdealiJK** [comment]
I think the python `logging` module may have this inbuilt and probably does it nicely

**Comment by sils1297** [comment]
blocked by https://github.com/coala-analyzer/coala/issues/975

**Comment by sils** [comment]
No longer blocked, we do use builtin logging now

**Comment by Makman2** [comment]
What about logging coala always instead of only on errors (which is a bit complicated, isn't it?)

**Comment by nemaniarjun** [comment]
No, Its not. @Makman2 This can be easily implemented. Assign me if still needed.

**Comment by nemaniarjun** [comment]
A bit of design needed, we need to write all the logs in a log file, in the place where coala is run, is that it?

**Comment by Makman2** [comment]
Yes would say so :+1: :)

**Comment by nemaniarjun** [comment]
Nearly done with this.
Do we want the file to be colored for console viewing or not? ( I think not, just confirming :P )

**Comment by Makman2** [comment]
No by default no colors.

**Comment by nemaniarjun** [comment]
Will send a PR as soon as the current one is merged. https://github.com/coala/coala/pull/3494

**Comment by jayvdb** [comment]
@nemaniarjun , what is the status of this. https://github.com/coala/coala/pull/3596 was closed, and a new PR was going to be opened ... ;-)

**Comment by nemaniarjun** [comment]
@jayvdb I was a bit busy with exams, etc. You can expect it to be completed by the end of this week.
Current PR : https://github.com/coala/coala/pull/3740

**Comment by jayvdb** [comment]
No worries; I have cross linked the PR with the issue so it doesnt get lost.

**Comment by nemaniarjun** [comment]
Creating a new PR. It is very rough as I am very rusty and this issue is / was complicated.
But looking forward to getting this done soon :)
